inc/class-core-sitemaps-index.php

```diff
@@ -14,35 +14,39 @@
  * Builds the sitemap index page that lists the links to all of the sitemaps.
  */
 class Core_Sitemaps_Index {
+
 	/**
-	 * Sitemap name.
-	 * Used for building sitemap URLs.
+	 * The main registry of supported sitemaps.
 	 *
-	 * @var string
+	 * @var Core_Sitemaps_Registry
 	 */
-	protected $name = 'index';
+	protected $registry;
 
 	/**
-	 * A helper function to initiate actions, hooks and other features needed.
+	 * Core_Sitemaps_Index constructor.
+	 *
+	 * @param Core_Sitemaps_Registry $registry Sitemap provider registry.
 	 */
-	public function setup_sitemap() {
-		// Add filters.
-		add_filter( 'robots_txt', array( $this, 'add_robots' ), 0, 2 );
-		add_filter( 'redirect_canonical', array( $this, 'redirect_canonical' ) );
+	public function __construct( $registry ) {
+		$this->registry = $registry;
 	}
 
 	/**
-	 * Prevent trailing slashes.
+	 * Gets a sitemap list for the index.
```

Comment:
Let's add `@since 5.5.0`

Reply:
Also applies to all other new methods, of course.

Reply:
Done.